### PR TITLE
(DOCSP 30002): Flutter FTS

### PR DESCRIPTION
**Pull request by lindseymoore**
### Jira

- https://jira.mongodb.org/browse/DOCSP-30002

### Staged Changes

- [Property Annotations](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-30002/sdk/flutter/realm-database/model-data/property-annotations/)
- [Read and Write Data](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-30002/sdk/flutter/realm-database/read-and-write-data/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [ ] Create Jira ticket for corresponding docs-app-services update(s), if any
- [ ] Checked/updated Admin API
- [ ] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/flutter/realm-database/model-data/property-annotations.txt [hunk 1]
```diff
@@ -111,3 +111,27 @@ The following data types can be indexed:
...
+Full-Text Search Indexes
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to standard indexes, Realm also supports Full-Text Search (FTS)
+indexes on string properties. While you can query a string field with or without 
+a standard index, an FTS index enables searching for multiple words and phrases 
+and excluding others.
+
+For more information on querying full-text indexes, see 
+:ref:`Filter with Full Text Search <flutter-fts>`.
```
Comment (krollins-mdb) [inline source/sdk/flutter/realm-database/model-data/property-annotations.txt:126]: Couple small edits for consistency in how we refer to FTS.
```suggestion
For more information on querying FTS indexes, see 
:ref:`Filter with Full-Text Search <flutter-fts>`.
```

File: source/sdk/flutter/realm-database/model-data/property-annotations.txt [hunk 2]
```diff
@@ -111,3 +111,27 @@ The following data types can be indexed:
...
+
+To create a FTS index on a property, use the :flutter-sdk:`@Indexed
```
Comment (krollins-mdb) [inline source/sdk/flutter/realm-database/model-data/property-annotations.txt:128]:
```suggestion
To create an FTS index on a property, use the :flutter-sdk:`@Indexed 
```

File: source/sdk/flutter/realm-database/model-data/property-annotations.txt [hunk 3]
```diff
@@ -111,3 +111,27 @@ The following data types can be indexed:
...
+In addition to standard indexes, Realm also supports Full-Text Search (FTS)
+indexes on string properties. While you can query a string field with or without 
+a standard index, an FTS index enables searching for multiple words and phrases 
+and excluding others.
+
+For more information on querying full-text indexes, see 
+:ref:`Filter with Full Text Search <flutter-fts>`.
+
+To create a FTS index on a property, use the :flutter-sdk:`@Indexed 
+<realm_common/Indexed-class.html>` annotation and specify the
```
Comment (krollins-mdb) [inline source/sdk/flutter/realm-database/model-data/property-annotations.txt:129]: TIL, `:flutter-sdk:` doesn't work for all of the Flutter SDK's docs. Looks like `realm_common`, which `Indexed` is part of, has a different root URL, so this convention results in a 404.

We'll need to use the full URL, like lines 94 and 95 in this file.

File: source/sdk/flutter/realm-database/model-data/property-annotations.txt [hunk 4]
```diff
@@ -111,3 +111,27 @@ The following data types can be indexed:
...
+indexes on string properties. While you can query a string field with or without 
+a standard index, an FTS index enables searching for multiple words and phrases 
+and excluding others.
+
+For more information on querying full-text indexes, see 
+:ref:`Filter with Full Text Search <flutter-fts>`.
+
+To create a FTS index on a property, use the :flutter-sdk:`@Indexed 
+<realm_common/Indexed-class.html>` annotation and specify the 
+:flutter-sdk:`IndexType <realm_common/RealmIndexType.html>` as ``fullText``.
```
Comment (krollins-mdb) [inline source/sdk/flutter/realm-database/model-data/property-annotations.txt:130]: Same as above. Right now, this 404s.

File: source/sdk/flutter/realm-database/model-data/property-annotations.txt [hunk 5]
```diff
@@ -111,3 +111,27 @@ The following data types can be indexed:
...
+and excluding others.
+
+For more information on querying full-text indexes, see 
+:ref:`Filter with Full Text Search <flutter-fts>`.
+
+To create a FTS index on a property, use the :flutter-sdk:`@Indexed 
+<realm_common/Indexed-class.html>` annotation and specify the 
+:flutter-sdk:`IndexType <realm_common/RealmIndexType.html>` as ``fullText``. 
+This enables full-text queries on the property. In the 
+following example, we mark the genre property with the FTS annotation:
```
Comment (krollins-mdb) [inline source/sdk/flutter/realm-database/model-data/property-annotations.txt:132]:
```suggestion
following example, we mark the pattern and material properties with the FTS annotation:
```

File: source/sdk/flutter/realm-database/read-and-write-data.txt [hunk 6]
```diff
@@ -75,6 +75,40 @@ For more information on constructing queries, refer to the
 .. literalinclude:: /examples/generated/flutter/read_write_data_test.snippet.filter.dart
    :language: dart
 
+.. _flutter-fts:
+
+Filter with Full-Text Search
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use Realm Query Language (RQL) to query on properties that have
+:ref:`Full-Text Search Indexes <flutter-fts-annotation>` (FTS) on them.
```
Comment (krollins-mdb) [inline source/sdk/flutter/realm-database/read-and-write-data.txt:84]:
```suggestion
You can use Realm Query Language (RQL) to query on properties that use
:ref:`Full-Text Search Index <flutter-fts-annotation>` annotations. 
```

**Review by krollins-mdb — CHANGES_REQUESTED** [review]
@lindseymoore, thank you so much for putting in the work on Full-Text Search!

I have some minor text suggestions and link fixes. We need a bit more work on the unit test, but you got most of the way there.

**Comment by krollins-mdb** [comment]
Also, Autobuilder is [mad about inconsistent title levels](https://github.com/mongodb/docs-realm/actions/runs/5350295998/jobs/9702710675?pr=2847), but I couldn't find anything wrong. *shrug* Maybe take a look and see if you can find something I missed.

File: examples/dart/test/full_text_search_test.dart
[the remark this answers is not in the document]
Comment (lindseymoore) [reply examples/dart/test/full_text_search_test.dart:1]: Thank you, this was very helpful!

**Pull request merged**